src/main/java/pl/agh/edu/iisg/io/vmms/vmmsbackend/controller/ReservationController.java

```diff
@@ -0,0 +1,233 @@
+package pl.agh.edu.iisg.io.vmms.vmmsbackend.controller;
+
+import org.modelmapper.ModelMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+import pl.agh.edu.iisg.io.vmms.vmmsbackend.dto.UserDto;
+import pl.agh.edu.iisg.io.vmms.vmmsbackend.dto.reservation.ReservationDto;
+import pl.agh.edu.iisg.io.vmms.vmmsbackend.dto.reservation.ReservationRequestDto;
+import pl.agh.edu.iisg.io.vmms.vmmsbackend.dto.reservation.ReservationResponseDto;
+import pl.agh.edu.iisg.io.vmms.vmmsbackend.exception.ReservationDateNotFoundException;
+import pl.agh.edu.iisg.io.vmms.vmmsbackend.exception.ReservationExpiredException;
+import pl.agh.edu.iisg.io.vmms.vmmsbackend.exception.ReservationNotFoundException;
+import pl.agh.edu.iisg.io.vmms.vmmsbackend.exception.http.HttpException;
+import pl.agh.edu.iisg.io.vmms.vmmsbackend.model.User;
+import pl.agh.edu.iisg.io.vmms.vmmsbackend.model.VMPool;
+import pl.agh.edu.iisg.io.vmms.vmmsbackend.model.reservations.Reservation;
+import pl.agh.edu.iisg.io.vmms.vmmsbackend.model.reservations.ReservationPeriod;
+import pl.agh.edu.iisg.io.vmms.vmmsbackend.service.ReservationService;
+import pl.agh.edu.iisg.io.vmms.vmmsbackend.service.UserService;
+import pl.agh.edu.iisg.io.vmms.vmmsbackend.service.VMPoolService;
+
+import javax.validation.Valid;
+import java.util.*;
+import java.util.stream.Collectors;
+
+@CrossOrigin("*")
+@RestController
+public class ReservationController {
+
+    private static final String RESERVATIONS_ENDPOINT = "/reservations";
+    private static final String RESERVATIONS_IN_PERIOD_ENDPOINT = "/reservations/between";
+    private static final String RESERVATION_ENDPOINT = "/reservation";
+    private static final String CONFIRM_TMP_ENDPOINT = "/reservation/confirm";
+    private static final String CANCEL_TMP_ENDPOINT = "/reservation/cancel";
+    private static final String DELETE_DATES_FROM_RESERVATION_ENDPOINT = "/reservation/dates";
```

Comment:
most of those static fields start with '/reservation' - maybe it would be better to make the whole controller's path start with such prefix?